### DISCUSSION TITLE
ADR DR_0 ABP but trying to send more than 11 bytes, uplink fail

### DISCUSSION THREAD

**Discussion by Tom1278** [Lora-net/LoRaMac-node#1057] [Q&A] [answered]
I've been trying for the last 3 hours to coerce the LoRa stack to send the uplink with DR 1 instead of DR 0. I'm trying to send 20 bytes.
ADR is on, using ABP.
Every time the stack overrides my command and switches back to DR 0. In the beginning it basically sends empty frames at DR 0 only when the gateway finally tells it to switch to DR 1 or I think it's actually telling it to go to DR 4, then it can finally send the data. My concern is that the LoRa device may not be able to receive the gateway or something else goes wrong and then it will be dead in the water not sending any data. What the heck do I have to do to tell this thing to at least start sending at DR 1 since I will never send less than 12 bytes in my payload?

I tried this:

    mibReq.Type = MIB_CHANNELS_DATARATE;
    mibReq.Param.ChannelsDatarate = DR_1;
    LoRaMacMibSetRequestConfirm( &mibReq );

    mibReq.Type = MIB_CHANNELS_DEFAULT_DATARATE;
    mibReq.Param.ChannelsDatarate = DR_1;
    LoRaMacMibSetRequestConfirm( &mibReq );

I also tried this:
#define LORAWAN_DEFAULT_DATARATE                    DR_1

**Comment by mluis1** [comment]
Could you please provide some more information?

- Which version of this repository are you using?
- Which version of LoRaWAN specification are you using?

As per LoRaWAN 1.0.4 and higher specifications the first uplink when operating in ABP mode with ADR bit ON shall be performed with default data rate (minimum data rate). Please refer to below specification excerpt:

![image](https://user-images.githubusercontent.com/5763295/110287875-7c07f000-7fe7-11eb-8b53-4caaa51a846e.png)

**Comment by MedadRufus** [comment, marked as answer]
This issue is similar to #1019
